src/cliffordsynthesis/CliffordSynthesizer.cpp

```diff
@@ -122,7 +121,7 @@ CliffordSynthesizer::determineUpperBound(EncoderConfig config) {
     return {0U, config.timestepLimit};
   }
 
-  std::size_t lowerBound = 0U;
+  std::size_t lowerBound = config.minimalTimesteps;
   std::size_t upperBound = config.timestepLimit;
```

Comment:
I believe a couple of things are still not right here:
 - At the top, in case a circuit is provided, the result should most likely be `{config.minimalTimesteps, config.timestepLimit}`.
 - This leads to another point that also affects the main iteration: it should be ensured that `upperBound` is at least as large as `lowerBound`. Otherwise this might produce nonsense.

I'd say it's propbably best to move such a sanity check in front of the `if(results.sat())` check, i.e., something like:
```c++
config.timestepLimit = std::max(config.timestepLimit, config.minimalTimesteps);
std::size_t lowerBound = config.minimalTimesteps;
std::size_t upperBound = config.timestepLimit;

if (results.sat()) {
  return {lower, upper};
}
```

Reply:
No worries. That's what these PR reviews are for.